web/src/markdown.js

```diff
@@ -388,7 +388,25 @@ function make_emoji_span(codepoint, title, alt_text) {
 }
 
 function handleUnicodeEmoji({unicode_emoji, get_emoji_name}) {
-    const codepoint = unicode_emoji.codePointAt(0).toString(16);
+    const emoji_codepoints = [];
+    for (let i = 0; i < unicode_emoji.length; i += 1) {
+        const char_codepoint = unicode_emoji.codePointAt(i).toString(16);
+        // TODO: I don't know why the ones starting with d need to be removed
+        if (char_codepoint.startsWith("d")) {
+            continue;
+        }
+        emoji_codepoints.push(char_codepoint);
+    }
+
+    let codepoint = emoji_codepoints.join("-");
+    // This unqualifies qualified emoji, which helps us make sure we
+    // can match both versions.
+    // We keep fe0f-20e3 because that pattern shows up in keycaps,
+    // which we leave qualified so that if the user types an unqualified
+    // keycap (e.g. 1 in a box), we don't turn it into an image.
+    if (codepoint.includes("fe0f") && !codepoint.includes("fe0f-20e3")) {
+        codepoint = codepoint.replace("fe0f", "");
```

Comment:
This will leave stray `-`s in the wrong places and incorrectly matches `1fe0f` or `fe0f1`. Do the replacement on the original string, not the hex conversion.

Reply:
Ah yeah -- okay figured out how to do it beforehand, thanks!